examples/v2_dry_run.rs

```diff
@@ -0,0 +1,117 @@
+//! Example of dry-run functionality of the node.
+
+use anyhow::Context;
+use clap::AppSettings;
+use concordium_base::{
+    base::Energy,
+    common::types::Timestamp,
+    contracts_common::{Address, Amount, ContractAddress, EntrypointName},
+    smart_contracts::{OwnedParameter, OwnedReceiveName},
+    transactions::Payload,
+};
+use concordium_rust_sdk::{
+    types::smart_contracts::ContractContext,
+    v2::{self, dry_run::DryRunTransaction, BlockIdentifier},
+};
+use structopt::StructOpt;
+
+#[derive(StructOpt)]
+struct App {
+    #[structopt(
+        long = "node",
+        help = "GRPC interface of the node.",
+        default_value = "http://localhost:20000"
+    )]
+    endpoint: v2::Endpoint,
+}
+
+/// Test all dry run operations.
+async fn test_all(endpoint: v2::Endpoint) -> anyhow::Result<()> {
+    // Connect to endpoint.
+    let mut client = v2::Client::new(endpoint).await.context("Cannot connect.")?;
+    // Start the dry run session.
+    let mut dry_run = client.dry_run().await?;
+    println!(
+        "Timeout: {:?}\nEnergy quota: {:?}",
+        dry_run.timeout(),
+        dry_run.energy_quota()
+    );
+    // Load the best block.
+    let fut1 = dry_run.load_block_state(BlockIdentifier::Best).await?;
+    // Load the last finalized block.
+    let fut2 = dry_run.load_block_state(BlockIdentifier::LastFinal).await?;
+    // Await the results of the loads in the reverse order.
+    let res2 = fut2.await?;
+    let res1 = fut1.await?;
```

Comment:
Do we really need to support having two requests running at a time? `res2` will not be available until `res1` is anyway, so it is just saving the 'reaction' time.
If not, then I don't think we need `SharedReceiver<S>` and can have `request` await for a response directly in the same future. This will also have the borrow checker prevent sending multiple requests.

Reply:
In line with this, I think the API that I would most likely want to use is this immediate return, in combination with a `send` based API where I can fire off all the requests and then get a stream of all responses for some niche uses.

Reply:
I have changed the default API to block for a response.